### DISCUSSION TITLE
Capture to variable file

### DISCUSSION THREAD

**Discussion by asmithja** [nvim-orgmode/orgmode#540] [Ideas]
Hello,
I would like my journal and meeting notes to be in separate files, so I created the following configuration
```
org_capture_templates   = {
  s = {
    description = 'Standup',
    template = '* Standup %<%d.%m.%Y> \n [....]',
    target = '[...]/standup/standup-%<%d.%m.%Y>.org'
  }
}
```
I then realized that `target` will not be parsed, so I monkey patched [capture/init.lua:134ll](https://github.com/nvim-orgmode/orgmode/blob/master/lua/orgmode/capture/init.lua#L134) to:
```
local fName = template.target
fName = self.templates:_compile_dates(fName)
fName = self.templates:_compile_expansions(fName)
fName = self.templates:_compile_expressions(fName)
fName = self.templates:_compile_prompts(fName)
local file = vim.fn.resolve(vim.fn.fnamemodify(fName or config.org_default_notes_file, ':p'))
```
I'm aware that starting a discussion for this is not the (final) way to go, but before I fork, change and pull request, I wanted to ask if the **idea behind the change** is acceptable for this project - I have NO idea how emacs orgmode would handle this (not using emacs).

(Additionally, I guess it would be better to refactor [Template:compile](https://github.com/nvim-orgmode/orgmode/blob/master/lua/orgmode/capture/templates.lua#L53), extracting the parts that would be redundant after above changes and instead call them from `Template.compile` and `Capture:_get_refile_vars()`)

Would you welcome said change?

**Comment by kristijanhusak** [comment]
Hi,

I guess this makes perfect sense. Go ahead and make a PR once you have something and we can go from there.

**Comment by sotte** [comment]
See https://github.com/nvim-orgmode/orgmode/pull/591

**Comment by hoclun-rigsep** [comment]
I would like this kind of substitution to work in the `headline` field too!
